### PR TITLE
SL-18721 Shutdown fixes

### PR DESCRIPTION
**Pull request by akleshchev**

### CODE REVIEW

**Review by marchcat-pe — APPROVED** [review]

File: indra/newview/llappviewermacosx-objc.h [hunk 1]
```diff
@@ -33,4 +33,6 @@
 //Why?  Because BOOL
 void launchApplication(const std::string* app_name, const std::vector<std::string>* args);
 
+void force_ns_sxeption() { [NSEXception raise:@"Forced NSException"] };
```
Comment (marchcat-pe) [inline indra/newview/llappviewermacosx-objc.h:36]: mac build looks unhappy about this:
`/Users/runner/work/viewer/viewer/indra/newview/llappviewermacosx-objc.h:36:41: error: expected ',' or ']' in lambda capture list`
also
`/Users/runner/work/viewer/viewer/indra/newview/llappviewermacosx-objc.h:36:29: error: use of undeclared identifier 'NSEXception'`

Comment (marchcat-pe) [inline indra/newview/llappviewermacosx-objc.h:36]:
```suggestion
void force_ns_sxeption() { [NSException raise:@"Forced NSException"] };
```
Mac build is unhappy. Probably due to typo.

Comment (akleshchev) [reply indra/newview/llappviewermacosx-objc.h:36]: Thank you, that's why it's a draft, will run it on mac a bit later.

**Review by maxim-productengine — APPROVED** [review]

**Review by marchcat-pe — APPROVED** [review]

**Comment by AtlasLinden** [comment]
Archived issue: https://github.com/secondlife/jira-archive-internal/issues/68974

**Pull request merged**